File: kos/kos-server/src/main/webapp/resources/js/product/category-service.js
(function(){
	var CATEGORY_URL = "/category";
	
	angular
		.module("kos")
		.factory("categoryService", CategoryService);
	
	function CategoryService($http){
		return {
			findByName : findByName,
			findRootCategories : findRootCategories
		};
		
		function findByName(name){
			return $http
				.get(CATEGORY_URL + "?name=" + name)
				.then(findByNameSuccess);
			
			function findByNameSuccess(response){
				return response.data;
			}
		}
		
		function findRootCategories(){
			return $http
				.get(CATEGORY_URL + "?rootCategory=true")
				.then(findRootCategoriesSuccess);
			
			function findRootCategoriesSuccess(response){
				return response.data;
			}
		}
	}
})();

File: kos/kos-server/src/main/java/com/cspinformatique/kubik/kos/domain/product/service/CategoryService.java
package com.cspinformatique.kubik.kos.domain.product.service;

import java.util.List;

import com.cspinformatique.kubik.kos.model.product.Category;
import com.cspinformatique.kubik.server.model.kos.KosNotification;

public interface CategoryService {	
	Category findByKubikId(int kubikId);
	
	Category findByName(String name);
	
	List<Category> findByRootCategory(boolean rootCategory);

	Category findOne(int id);
	
	List<Category> generateNestedCategories(int categoryId);

	void processKosNotification(KosNotification kosNotification);
}
